src/main/java/teammates/storage/search/FeedbackResponseCommentSearchDocument.java

```diff
@@ -255,69 +214,47 @@ public static FeedbackResponseCommentSearchResultBundle fromResults(
         List<ScoredDocument> filteredResults = filterOutCourseId(results, instructors);
         for (ScoredDocument doc : filteredResults) {
             // get FeedbackResponseComment from results
-            FeedbackResponseCommentAttributes comment = JsonUtils.fromJson(
-                    doc.getOnlyField(Const.SearchDocumentField.FEEDBACK_RESPONSE_COMMENT_ATTRIBUTE).getText(),
-                    FeedbackResponseCommentAttributes.class);
-            FeedbackResponseCommentAttributes commentFromDb = frcDb.getFeedbackResponseComment(comment.getId());
-            if (commentFromDb == null) {
-                frcDb.deleteDocument(comment);
+            long feedbackResponseCommentId = Long.parseLong(doc.getId());
+            FeedbackResponseCommentAttributes comment = frcDb.getFeedbackResponseComment(feedbackResponseCommentId);
+            if (comment == null) {
+                // search engine out of sync
```

Comment:
Is it a (big) problem if this happens?

Reply:
Search engine out of sync as SearchManager may fail to delete documents due to GAE error. The chance is low and it is generally not a big problem.

Comment also has been added to explain this.